src/common/message-creators/bug-action-message-creator.ts

```diff
@@ -1,37 +1,32 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 import { BaseActionPayload } from '../../background/actions/action-payloads';
-import { Message } from '../message';
 import { Messages } from '../messages';
 import { TelemetryDataFactory } from '../telemetry-data-factory';
 import { FILE_ISSUE_CLICK, FileIssueClickService, TelemetryEventSource } from '../telemetry-events';
-import { BaseActionMessageCreator } from './base-action-message-creator';
+import { ActionMessageDispatcher } from './action-message-dispatcher';
 
-export class BugActionMessageCreator extends BaseActionMessageCreator {
+export class BugActionMessageCreator {
     constructor(
-        postMessage: (message: Message) => void,
-        tabId: number,
+        private readonly dispatcher: ActionMessageDispatcher,
```

Comment:
Not blocking: we could make the other constructor parameters here `readonly` too, yes?

Reply:
Make sense. Updated.